### ISSUE TITLE
False positive :missing-docstring on `deftype` and `deftest`

### ISSUE THREAD

**Issue by serioga** [borkdude/clj-kondo#800]
```clojure
(ns foo
  {:clj-kondo/config {:linters {:missing-docstring {:level :warning}}}}
  (:require [clojure.test :as t]))
(deftype Foo [x])
(t/deftest foo)
```
> warning: Missing docstring.

**Comment by borkdude** [comment]
in carve we ignore vars that are created by deftype / deftest, etc like this:
https://github.com/borkdude/carve/blob/0e2a6014e684b2f2cea9e95e44d7bab215ba904f/src/carve/impl.clj#L108-L118

**Comment by serioga** [comment]
about `rum` macros and missing docstrings.

```
(defcs name doc-string? (< mixins+)? [ state-arg params* ] render-body+)
```

when I define component like

```clojure
(rum/defcs hello-world
  "Example react component."
  [state name]
  "foo")
```

then `:missing-docstring` linter still compains about missing docstring.

corresponding configuration:
```clojure
{:linters {:missing-docstring {:level :warning}}
 :lint-as {rum.core/defc clj-kondo.lint-as/def-catch-all
           rum.core/defcs clj-kondo.lint-as/def-catch-all
           rum.core/defcc clj-kondo.lint-as/def-catch-all}}
```

**Comment by borkdude** [comment]
This issue returned when fixing #833.

**Comment by serioga** [comment]
looks like similar issue for `cljs.core/defrecord`...

**Comment by borkdude** [comment]
New issue please
